backend/controllers/addAppointmentDetails.js

```diff
@@ -23,6 +23,67 @@ const setSelected = (value, items) =>
         : item
   )
 
+const buildDateTime = ({ date, hours, minutes }) => {
+  const time = date && hours && minutes && moment(date, DAY_MONTH_YEAR)
+  return time && time.hour(Number(hours)).minutes(Number(minutes))
+}
+
+const extractHoursMinutes = dateTime => {
+  const instant = moment(dateTime, DATE_TIME_FORMAT_SPEC)
+  const hours = instant.hours()
+  const minutes = instant.minutes()
+
+  return {
+    hours: hours <= 9 ? `0${hours}` : hours,
+    minutes: minutes <= 9 ? `0${minutes}` : minutes,
+  }
+}
+
+const getValidationMessages = ({
+  appointmentType,
+  location,
+  date,
+  startTime,
+  endTime,
+  sameTimeAppointments,
+  comments,
+}) => {
+  const errors = []
+  const now = moment()
+  const isToday = date ? moment(date, DAY_MONTH_YEAR).isSame(now, 'day') : false
+
+  if (!appointmentType) errors.push({ text: 'Select an appointment type', href: '#appointment-type' })
+  if (!location) errors.push({ text: 'Select a location', href: '#location' })
+  if (!date) errors.push({ text: 'Select a date', href: '#date' })
+  if (!sameTimeAppointments)
+    errors.push({ text: 'Select yes if the appointments all have the same time', href: '#same-time-appointments' })
+
+  if (date && !moment(date, DAY_MONTH_YEAR).isValid())
+    errors.push({ text: 'Enter a date in DD/MM/YYYY format', href: '#date' })
+
+  if (date && moment(date, DAY_MONTH_YEAR).isBefore(now, 'day'))
+    errors.push({ text: 'Select a date that is not in the past', href: '#date' })
+
+  if (sameTimeAppointments === 'yes') {
+    const startTimeDuration = moment.duration(now.diff(startTime))
+    const endTimeDuration = endTime && moment.duration(startTime.diff(endTime))
+
+    if (!startTime) errors.push({ text: 'Select a start time', href: '#start-time-hours' })
+
+    if (isToday && startTimeDuration.asMinutes() > 1)
+      errors.push({ text: 'Select a start time that is not in the past', href: '#start-time-hours' })
+
+    if (endTime && endTimeDuration.asMinutes() > 1) {
+      errors.push({ text: 'Select an end time that is not in the past', href: '#end-time-hours' })
+    }
+  }
+
+  if (comments && comments.length > 3600)
```

Comment:
out of curiosity why are you doing this manually rather than using the https://design-system.service.gov.uk/components/character-count/ component (instead of textarea)?

Reply:
Andy H didn't want to use that component due to the fear of creating too much noise.

Reply:
:astonished: